src/api.rs

```diff
@@ -754,17 +784,47 @@ impl<T: Pixel> Context<T> {
     ))
   }
 
+
+  /// Send the information to the encoder
+  ///
+  /// ``` no_run
+  /// use rav1e::prelude::*;
+  ///
+  /// let cfg = Config::default();
+  /// let mut ctx: Context<u8> = cfg.new_context();
+  /// let f1 = ctx.new_frame();
+  /// let f2 = f1.clone();
+  /// let info = FrameParameters { keyframe : true };
+  ///
+  /// // Send the plain frame data
+  /// ctx.send_frame(f1);
+  /// // Send the data and the per-frame parameters
+  /// // In this case the frame is forced to be a keyframe.
+  /// ctx.send_frame((f2, info));
+  /// // Flush the encoder, it is equivalent to a call to `flush()`
+  /// ctx.send_frame(None);
+  /// ```
+  ///
+  /// It could return `EncoderStatus::EnoughData` if the incoming
+  /// frame queue is full or `EncoderStatus::Failure` if the per-frame
+  /// parameters set an impossible constraint.
   pub fn send_frame<F>(&mut self, frame: F) -> Result<(), EncoderStatus>
   where
-    F: Into<Option<Arc<Frame<T>>>>
+    F: IntoFrame<T>
   {
-    let frame = frame.into();
+    let (frame, props) = frame.into();
```

Comment:
We should be consistent in the use of props vs params.

Reply:
naming things is hard: I let you pick properties vs params.

Reply:
Params makes more sense to me, since we're setting how to encode it, not the inherent properties of the frame (like colorspace).

Reply:
Done.